### PR TITLE
Update coupon search to allow searching by code

### LINKED ISSUE #1611: Question: Coupons Endpoint - should search include description?
**Describe the bug**

Found as part of #1525.

The coupons report endpoint `search` parameter searches the coupon description - should it?

This can lead to somewhat confusing results since the description isn't shown in the report table.

**To Reproduce**

1. Go to Coupons Report
2. Search for a term that's in a coupon description, but not it's code (like "woo" on WCCOM data)
3. See results that don't appear to contain the search term

**Expected behavior**

Perhaps the `199...` coupon shouldn't be there? _(See screenshot below)_

**Screenshots**

screen shot 2019-02-18 at 1 03 39 pm

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #1611 

Searches coupons by code to avoid mismatched results by description and blocking result sets > 10.

### Before
Screen Shot 2019-03-14 at 3 17 26 PM

### After
Screen Shot 2019-03-14 at 3 18 09 PM

### Detailed test instructions:

1.  Add a coupon with a keyword in the description, but not in the coupon code (title).
2.  Search for that keyword under the coupons table.
3.  Check that the result is not returned.
4.  Check that the results in the network request for the autocompleter also do not return that coupon.

### CODE REVIEW

**Review by emdashcodes — APPROVED** [review]
Code looks good, and this is testing well for me. 👍

File: includes/api/class-wc-admin-rest-coupons-controller.php [hunk 1]
```diff
@@ -24,4 +24,69 @@ class WC_Admin_REST_Coupons_Controller extends WC_REST_Coupons_Controller {
...
+	 * @param object $wp_query WP_Query object.
+	 * @return string
+	 */
+	public static function add_wp_query_search_code_filter( $where, $wp_query ) {
+		global $wpdb;
+
+		$code_search = $wp_query->get( 'search_code' );
+		if ( $code_search ) {
+			$code_search = $wpdb->esc_like( $code_search );
+			$code_search = ' \'%' . $code_search . '%\'';
```
Comment (jeffstieler) [inline includes/api/class-wc-admin-rest-coupons-controller.php]: Nitpick: using `"` here would avoid escape characters.

File: packages/components/src/search/autocompleters/coupons.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ export default {
 		let payload = '';
 		if ( search ) {
 			const query = {
-				search,
+				search_code: search,
```
Comment (jeffstieler) [inline packages/components/src/search/autocompleters/coupons.js]: Can we avoid renaming this? _I sound like a broken record..._

**Review by jeffstieler — COMMENTED** [review]
This tests well, just some questions/comments.

File: packages/components/src/search/autocompleters/coupons.js (hunk 2 above)
Comment (joshuatf) [reply packages/components/src/search/autocompleters/coupons.js]: No problemo- done!

File: includes/api/class-wc-admin-rest-coupons-controller.php (hunk 1 above)
Comment (joshuatf) [reply includes/api/class-wc-admin-rest-coupons-controller.php]: Added double quotes here as well as other PRs that override search params.

**Pull request merged**